lighthouse-core/gather/fetcher.js

```diff
@@ -216,17 +224,18 @@ class Fetcher {
           return;
         }
 
-        // Now in the response stage, but the request failed.
-        if (!(responseStatusCode >= 200 && responseStatusCode < 300)) {
-          reject(new Error(`Invalid response status code: ${responseStatusCode}`));
-          return;
-        }
-
-        const responseBody = await this.session.sendCommand('Fetch.getResponseBody', {requestId});
-        if (responseBody.base64Encoded) {
-          resolve(Buffer.from(responseBody.body, 'base64').toString());
+        if (responseStatusCode >= 200 && responseStatusCode <= 299) {
```

Comment:
in the same way, it seems like we could still run `Fetch.getResponseBody` here too even if it was an error status code (as we've increasingly found, people love to put real bodies with error status codes :)

Reply:
The previous behavior was to throw on error status codes, I'm ok with changing the behavior though.

Reply:
https://github.com/GoogleChrome/lighthouse/pull/12423#discussion_r623384603 Because `Network.loadNetworkResource` does not provide a stream handle if there is an error status code, I think we need to keep this check so the behavior is consistent between the iframe and protocol fetchers.

Reply:
yes, much better to have the two versions consistent